lib/order.rb

```diff
@@ -0,0 +1,17 @@
+class Order
+  attr_reader :total, :as_hash
+
+  def initialize(order, total)
+    @total = total
+    @as_hash = parse order
+  end
+
+  private
+
+  def parse order
+    order.split(', ').reduce({}) do |result, item|
```

Comment:
Style/CollectionMethods: Prefer `inject` over `reduce`. (https://github.com/bbatsov/ruby-style-guide#map-find-select-reduce-size)<br>Style/EachWithObject: Use `each_with_object` instead of `reduce`.